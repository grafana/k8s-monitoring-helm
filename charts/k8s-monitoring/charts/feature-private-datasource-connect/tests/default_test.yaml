# yamllint disable rule:document-start rule:line-length rule:trailing-spaces
suite: Test default values
templates:
  - deployment.yaml
  - secret.yaml
tests:
  - it: should not create resources when PDC agent is disabled
    asserts:
      - hasDocuments:
          count: 0

  - it: should create deployment when PDC agent is enabled
    set:
      enabled: true
      credentials.createSecret: true
      credentials.token: "test-token"
      credentials.hostedGrafanaId: "test-grafana-id"
      credentials.cluster: "test-cluster"
    template: deployment.yaml
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: metadata.name
          value: release-name-feature-private-datasource-connect
      - matchSnapshot:
          path: spec

  - it: should create secret when PDC agent is enabled and createSecret is true
    set:
      enabled: true
      credentials.createSecret: true
      credentials.token: "test-token"
      credentials.hostedGrafanaId: "test-grafana-id"
      credentials.cluster: "test-cluster"
    template: secret.yaml
    asserts:
      - isKind:
          of: Secret
      - equal:
          path: metadata.name
          value: release-name-feature-private-datasource-connect
      - matchSnapshot:
          path: data
