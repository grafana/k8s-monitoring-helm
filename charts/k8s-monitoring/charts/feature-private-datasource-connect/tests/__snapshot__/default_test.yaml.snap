should create deployment when PDC agent is enabled:
  1: |
    replicas: 1
    selector:
      matchLabels:
        app.kubernetes.io/component: pdc-agent
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: pdc-agent
    template:
      metadata:
        labels:
          app.kubernetes.io/component: pdc-agent
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: pdc-agent
      spec:
        affinity:
          podAntiAffinity:
            preferredDuringSchedulingIgnoredDuringExecution:
              - podAffinityTerm:
                  labelSelector:
                    matchLabels:
                      app.kubernetes.io/component: pdc-agent
                      app.kubernetes.io/instance: RELEASE-NAME
                      app.kubernetes.io/name: pdc-agent
                  topologyKey: kubernetes.io/hostname
                weight: 100
        containers:
          - args:
              - -token
              - $(GCLOUD_PDC_SIGNING_TOKEN)
              - -cluster
              - $(GCLOUD_PDC_CLUSTER)
              - -gcloud-hosted-grafana-id
              - $(GCLOUD_HOSTED_GRAFANA_ID)
            env:
              - name: GCLOUD_PDC_SIGNING_TOKEN
                valueFrom:
                  secretKeyRef:
                    key: token
                    name: release-name-feature-private-datasource-connect
              - name: GCLOUD_HOSTED_GRAFANA_ID
                valueFrom:
                  secretKeyRef:
                    key: hosted-grafana-id
                    name: release-name-feature-private-datasource-connect
              - name: GCLOUD_PDC_CLUSTER
                valueFrom:
                  secretKeyRef:
                    key: cluster
                    name: release-name-feature-private-datasource-connect
            image: grafana/pdc-agent:latest
            imagePullPolicy: IfNotPresent
            name: pdc-agent
            resources:
              limits:
                cpu: 200m
                memory: 1Gi
              requests:
                cpu: 100m
                memory: 512Mi
should create secret when PDC agent is enabled and createSecret is true:
  1: |
    cluster: dGVzdC1jbHVzdGVy
    hosted-grafana-id: dGVzdC1ncmFmYW5hLWlk
    token: dGVzdC10b2tlbg==
